libobs/util/c99defs.h

```diff
@@ -51,6 +51,31 @@
 #define EXPORT __attribute__((visibility("default")))
 #endif
 
+#ifdef _MSC_VER
+#define PRAGMA_WARN_PUSH __pragma(warning(push))
+#define PRAGMA_WARN_POP __pragma(warning(pop))
+#define PRAGMA_WARN_DEPRECATION
+#define PRAGMA_WARN_STRINGOP_OVERFLOW
+#elif defined(__clang__)
+#define PRAGMA_WARN_PUSH _Pragma("clang diagnostic push")
+#define PRAGMA_WARN_POP _Pragma("clang diagnostic pop")
+#define PRAGMA_WARN_DEPRECATION \
+	_Pragma("clang diagnostic warning \"-Wdeprecated-declarations\"")
+#define PRAGMA_WARN_STRINGOP_OVERFLOW
+#elif defined(__GNUC__)
+#define PRAGMA_WARN_PUSH _Pragma("GCC diagnostic push")
+#define PRAGMA_WARN_POP _Pragma("GCC diagnostic pop")
+#define PRAGMA_WARN_DEPRECATION \
+	_Pragma("GCC diagnostic warning \"-Wdeprecated-declarations\"")
+#define PRAGMA_WARN_STRINGOP_OVERFLOW \
+	_Pragma("GCC diagnostic warning \"-Wstringop-overflow\"")
+#else
+#define PRAGMA_WARN_PUSH
+#define PRAGMA_WARN_POP
+#define PRAGMA_WARN_DEPRECATION
+#define PRAGMA_WARN_STRINGOP_OVERFLOW
+#endif
```

Comment:
Should these maybe have an `OBS_` prefix?

Reply:
I think it's better this way as they're closer to the standard `#pragma` syntax.